src/java/org/apache/cassandra/config/DataStorage.java

```diff
@@ -0,0 +1,385 @@
+     * @return a {@code DataStorage}
+     */
+    public static DataStorage inMegabytes(long megabytes)
+    {
+        return new DataStorage(megabytes, DataStorageUnit.MEGABYTES);
+    }
+
+    /**
+     * Returns the data storage unit.
+     *
+     * @return the data storage unit.
+     */
+    public DataStorageUnit getUnit()
+    {
+        return unit;
+    }
+
+    /**
+     * Returns the amount of data storage in bytes
+     *
+     * @return the amount of data storage in bytes
+     */
+    public long toBytes()
+    {
+        return unit.toBytes(quantity);
+    }
+
+    /**
+     * Returns the amount of data storage in bytes as an {@code int}
+     *
+     * @return the amount of data storage in bytes or {@code Integer.MAX_VALUE} if the number of bytes is too large.
+     */
+    public int toBytesAsInt()
+    {
+        return Ints.saturatedCast(toBytes());
+    }
+
+    /**
+     * Returns the amount of data storage in kilobytes
+     *
+     * @return the amount of data storage in kilobytes
+     */
+    public long toKilobytes()
+    {
+        return unit.toKilobytes(quantity);
+    }
+
+    /**
+     * Returns the amount of data storage in kilobytes as an {@code int}
+     *
+     * @return the amount of data storage in kilobytes or {@code Integer.MAX_VALUE} if the number of kilobytes is too large.
+     */
+    public int toKilobytesAsInt()
+    {
+        return Ints.saturatedCast(toKilobytes());
+    }
+
+    /**
+     * Returns the amount of data storage in megabytes
+     *
+     * @return the amount of data storage in megabytes
+     */
+    public long toMegabytes()
+    {
+        return unit.toMegabytes(quantity);
+    }
+
+    /**
+     * Returns the amount of data storage in megabytes as an {@code int}
+     *
+     * @return the amount of data storage in megabytes or {@code Integer.MAX_VALUE} if the number of megabytes is too large.
+     */
+    public int toMegabytesAsInt()
+    {
+        return Ints.saturatedCast(toMegabytes());
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(unit.toKilobytes(quantity));
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj)
+            return true;
+
+        if (!(obj instanceof DataStorage))
+            return false;
+
+        DataStorage other = (DataStorage) obj;
+        if (unit == other.unit)
+            return quantity == other.quantity;
+
+        // Due to overflows we can only guarantee that the 2 storages are equals if we get the same results
+        // doing the convertion in both directions.
+        return unit.convert(other.quantity, other.unit) == quantity && other.unit.convert(quantity, unit) == other.quantity;
+    }
+
+    @Override
+    public String toString()
+    {
+        return quantity + unit.symbol;
+    }
+
+    public String quantityToString()
+    {
+        return String.valueOf(quantity);
+    }
+
+    public enum DataStorageUnit
```

Comment:
can we merge `DataStorageUnit` with `BitRateUnit`?  You are just converting between size units and `BitRateUnit` has nothing to do with time.

Reply:
We can and then we can ensure no deviation in the units as you observed but for clarity was better to have them separately. I will look into this

Reply:
Yeah, at least if we go off the original issue description, it looks like we want to more or less remove `BitRateUnit`, given we explicitly didn't want to support Mbps, etc.

Reply:
On second thought, I guess it might make sense to keep `BitRateUnit`, because we do need to keep `toMegabitsPerSeconds()` and friends for the cases like `inter_dc_stream_throughput_outbound` where we'll need to convert the IEC unit in `DatabaseDescriptor.getInterDCStreamThroughputOutboundMegabitsPerSec()`.

Reply:
So to summarize, I think we want `BitRateUnit` to support just the IEC units (KiB, etc.), but then we add the proper converter *from* megabits/s and keep `toMegabitsPerSeconds()` et al.

Reply:
It seems we are on the same page. I agree with your last comment.